app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItem.kt

```diff
@@ -86,106 +93,163 @@ fun MessageItem(
+                                messageContent = messageContent,
+                                audioMessagesState = audioMessagesState,
+                                onAudioClick = onAudioClick,
+                                onChangeAudioPosition = onChangeAudioPosition,
+                                onAssetClick = currentOnAssetClicked,
+                                onImageClick = currentOnImageClick,
+                                onLongClick = onLongClick,
+                                onOpenProfile = onOpenProfile
+                            )
+                            MessageFooter(
+                                messageFooter,
+                                onReactionClicked
+                            )
+                        } else {
+                            MessageDecryptionFailure(
+                                messageHeader = messageHeader,
+                                decryptionStatus = messageHeader.messageStatus as MessageStatus.DecryptionFailure,
+                                onResetSessionClicked = onResetSessionClicked
+                            )
+                        }
+                    }
 
-                if (message.sendingFailed) {
-                    MessageSendFailureWarning(messageHeader.messageStatus as MessageStatus.MessageSendFailureStatus)
+                    if (message.sendingFailed) {
+                        MessageSendFailureWarning(messageHeader.messageStatus as MessageStatus.MessageSendFailureStatus)
+                    }
                 }
             }
         }
     }
 }
 
 @Composable
-private fun Modifier.customizeMessageBackground(
-    message: UIMessage
-) = run {
-    if (message.sendingFailed || message.receivingFailed) {
-        background(MaterialTheme.wireColorScheme.messageErrorBackgroundColor)
-    } else this
+fun MessageExpireLabel(messageContent: UIMessageContent?, timeLeft: String) {
+    when (messageContent) {
+        is UIMessageContent.TextMessage -> {
+            StatusBox(statusText = "Self-deleting message •  $timeLeft")
+        }
+
+        is UIMessageContent.AssetMessage -> {
+            StatusBox(
+                statusText = "Self-deleting message " +
+                        if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY) "•$timeLeft" else ""
+            )
+        }
+
+        is UIMessageContent.AudioAssetMessage -> {
+            StatusBox(
+                statusText = "Self-deleting message " +
+                        if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY) "•$timeLeft" else ""
+            )
+        }
+
+        is UIMessageContent.ImageMessage -> {
+            StatusBox(
+                statusText = "Self-deleting message" +
+                        " ${if (messageContent.downloadStatus == Message.DownloadStatus.SAVED_INTERNALLY) "•$timeLeft" else ""}"
+            )
+        }
+
+        else -> {}
+    }
```

Comment:
Maybe we could refactor all these items into another sealed class like `UIAssetMessageContent` so that they share at least the `assetId`, `downloadStatus` and `uploadStatus`. 
That way we can group them into easier `when` clauses.

Reply:
Will do in a seperate PR later on